File: dotnet/GHIElectronics.Due_/Pwm.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GHIElectronics.DUE {
    public partial class DUEController {
        public class PwmController {

            SerialInterface serialPort;

            public PwmController(SerialInterface serialPort) => this.serialPort = serialPort;

            public int Fixed_Frequency { get; } = 50;


            public bool Set(int pin, int dutycyle) {
                if (pin < 0 || pin >= MAX_IO)
                    throw new ArgumentOutOfRangeException("Invalid pin.");

                if (dutycyle < 0 || dutycyle > 1000) {
                    throw new Exception("Dutycle must be in 0..1000");
                }

                var cmd = string.Format("awrite({0},{1})", pin.ToString(), dutycyle.ToString());
     

                this.serialPort.WriteCommand(cmd);

                var res = this.serialPort.ReadRespone();

                if (res.success) {
                    return true;
                }

                return false;
            }

        }
    }
}

File: python/DUE/Pwm.py
from DUE.SerialInterface import SerialInterface

class PwmController:

    def __init__(self, serialPort:SerialInterface):
        self.serialPort = serialPort
        self.Fixed_Frequency = 50

    def Set(self, pin, duty_cycle):
        if pin < 0 or pin >= self.serialPort.DeviceConfig.MaxPinIO:
            raise ValueError('Invalid pin')

        if duty_cycle < 0 or duty_cycle > 1000:
            raise ValueError('Duty cycle must be in the range 0..1000')

        cmd = f'awrite({pin}, {duty_cycle})'
        self.serialPort.WriteCommand(cmd)

        res = self.serialPort.ReadRespone()
        if res.success:
            return True

        return False
